app/models/mixins/per_ems_worker_mixin.rb

```diff
@@ -101,10 +101,19 @@ def queue_name_for_ems(ems)
     end
 
     def ems_id_from_queue_name(queue_name)
-      return nil if queue_name.blank?
-      name, id = queue_name.split("_")
-      return nil unless name == "ems"
-      id.to_i
+      if queue_name.kind_of?(Array)
+        ems_ids = []
+        ems_ids.each do |ems_id|
```

Comment:
Should be `queue_name` instead of `ems_ids` here ... I couldn't figure out why this was always returning an empty array.

Also, this isn't converting the `id` to an int.

Though, leaning towards smaller methods, maybe something like this is better anyway:

```ruby
def parse_ems_id(queue_name)
  return nil if queue_name.blank?                                                                                                                                    
  name, id = queue_name.split("_")                                                                                                                                   
  return nil unless name == "ems"                                                                                                                                    
  id.to_i
end

def ems_id_from_queue_name(queue_name)
  queue_names = [queue_name] unless queue_name.kind_of?(Array)
  queue_name.collect { |q| parse_ems_id(q) }.flatten
end
```

This way, the parsing of the `ems_id` is done in one place one way, not in two places in the `if` and `else` block, slightly differently.

The only thing I don't like about this is that `ems_id_from_queue_name` would always handle arrays, but the name is singular.  But, I'm afraid to break the interface for that.